src/solver/conv_mlir_igemm_fwd.cpp

```diff
@@ -87,14 +66,6 @@ bool ConvMlirIgemmFwd::IsApplicable(const ConvolutionContext& ctx) const
     if(!ctx.IsFp32() && !ctx.IsFp16())
```

Comment:
This should also be moved into MLIR

Reply:
Conceptually all applicability check can be moved to let mlir handle it. However, that means every time MIOpen check for applicability of any config (like bf16), will ask MLIR's opinion. And the performance overhead of that check would be lowering through the first 3 steps of the pipeline.

My suggestion is that we try to hit at a balance between effectiveness and performance. Right now let's leave those extremely cheap checks in MIOpen untouched. Once we make the query overhead in checking with MLIR much lower than now, we can consider to remove those.